redis_json/src/ivalue_manager.rs

```diff
@@ -740,4 +744,32 @@ mod tests {
         let res = manager.get_memory(&value).unwrap();
         assert_eq!(res, 903);
     }
+
+    /// Tests the deserialiser of IValue for a string with unicode
+    /// characters, to ensure that the deserialiser can handle
+    /// unicode characters well.
+    #[test]
+    fn test_unicode_characters() {
+        let _guard = single_thread_test_mutex.lock();
```

Comment:
What resource needs to be mutual exclusive here?

Reply:
The `ijson` string interning implementation. It is hidden deep inside the implementation. Guy Korland patched it so that we don't use a thread-safe string interning simply because we don't need it, as it happens all within a single thread anyway. But the tests run concurrently, and so there is a race condition. There are different ways to solve this, one is just to run the tests with single thread, another is this. I picked this one because no one else needs to know about it from the outside, as it is only in the tests, and we actually don't have unit tests at all. The less people know and the less they are required to know about "how the project needs to be tested" - the better. Otherwise, we will have to change all our makefiles and everything else to run `cargo test` with just one thread, list it everywhere for everyone.... The people who are familiar with Rust, and people in general, in my opinion, should be able to run the tests with just `cargo test`. Everything else is hidden behind the implementation. By the way, we also do it this way in RedisGears, for example. So, let's consider this a just a unit-test implementation detail, if you don't mind :-)